.github/workflows/docs.yml

```diff
@@ -0,0 +1,28 @@
+name: build docs
+
+on:
+  pull_request:
+    branches:
+      - "*"
+  workflow_dispatch:
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: false
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.7
+          architecture: x64
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --requirement docs/requirements.txt
+      - name: Build docs
+        run: python -m mkdocs build --clean --site-dir artifacts/docs --config-file mkdocs.yml --strict

```

Comment:
This is fine, but maybe there's an existing mkdocs action that we could use?

Reply:
Most appear to be geared toward publishing the output of mkdocs. I did find Tiryoh/actions-mkdoc, which I pushed to my fork and ran a couple of builds. Check out 12 and 13 at https://github.com/blairconrad/FakeItEasy/actions. 12 uses mkdocs 1.3.0 and 13 uses 1.2.3. Both pass, I think because the docker image that the action uses has an old version of jinja2. 
As a result, I'm not sure we're replicating what the readthedocs build is doing, so I'm not super confident in it.

And it seems overly complicated to me, to be honest: deploy a docker container with mkdocs, which downgrades mkdocs from 1.3 to 1.2.3, then possibly reupgrade it…

Also, there's a weird ignored error I don't love:

```
ValueError

  Could not find a matching version of package ==

  at /root/.poetry/lib/poetry/console/commands/init.py:367 in _find_best_version_for_package
      363│         )
      364│ 
      365│         if not package:
      366│             # TODO: find similar
    → 367│             raise ValueError(
      368│                 "Could not find a matching version of package {}".format(name)
      369│             )
      370│ 
      371│         return package.pretty_name, selector.find_recommended_require_version(package)
```

I'm not sure this is the way to go. I can keep looking for another available just-build-the-mkdocs action, if you like.

Reply:
OK then, if it doesn't make things simpler, it's not worth it. Looks good to me as it is!

Reply:
https://github.com/jnanadarshan/mkdocs-build-action I think requires the requirements.txt file to be at the root, which is a possibility, but may break readthedocs. Or maybe we just put in the docs directory because it's tidier…